tools/cldr-code/src/main/java/org/unicode/cldr/test/CheckDates.java

```diff
@@ -641,6 +643,99 @@ && isTooMuchWiderThan(value, abbrValue)
         return this;
     }
 
+    static final Set<Integer> expectedField =
+            Set.of(
+                    DateTimePatternGenerator.YEAR,
+                    DateTimePatternGenerator.QUARTER,
+                    DateTimePatternGenerator.MONTH,
+                    DateTimePatternGenerator.DAY,
+                    DateTimePatternGenerator.WEEKDAY,
+                    DateTimePatternGenerator.HOUR,
+                    DateTimePatternGenerator.MINUTE,
+                    DateTimePatternGenerator.SECOND,
+                    DateTimePatternGenerator.ZONE);
```

Comment:
```suggestion
                    DateTimePatternGenerator.YEAR,
                    DateTimePatternGenerator.QUARTER,
                    DateTimePatternGenerator.MONTH,
                    DateTimePatternGenerator.WEEKDAY,
                    DateTimePatternGenerator.DAY,
                    DateTimePatternGenerator.HOUR,
                    DateTimePatternGenerator.MINUTE,
                    DateTimePatternGenerator.SECOND,
                    DateTimePatternGenerator.ZONE);
```

put in order

Reply:
Set.of makes an unordered immutable set (Hash). ImmutableSet.of makes an immutable ordered set (LinkedHash). While generally I prefer the latter but the order, for the test, doesn't matter. 

However we want weekday to be between day and hour, because in formatting you never say 

September, Tuesday, 1.
but rather
September 1, Tuesday